### PR TITLE
FEATURE: site settings to revoke api keys older than a number of days

### PR DESCRIPTION
**Pull request by renato**
`revoke api keys days` renamed to `revoke api keys unused days` and `expire user api keys days` renamed to `revoke user api keys unused days` to be more explicit about their actual behavior.

Two new site settings created: `revoke api keys maxlife days` and `revoke user api keys maxlife days` to respectively revoke api keys and user api keys created before a set number of days.

### CODE REVIEW

File: app/jobs/scheduled/clean_up_unused_user_api_keys.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+module Jobs
+  class CleanUpUnusedUserApiKeys < ::Jobs::Scheduled
+    every 1.day
+
+    def execute(args)
+      if SiteSetting.revoke_user_api_keys_unused_days > 0
+        revoke_days_ago = SiteSetting.revoke_user_api_keys_unused_days.days.ago
+
+        UserApiKey.where("last_used_at < ?", revoke_days_ago).update_all(revoked_at: Time.zone.now)
```
Comment (ZogStriP) [inline app/jobs/scheduled/clean_up_unused_user_api_keys.rb:11]: Should we also test (filter) that revoked_at is NULL?

File: spec/jobs/clean_up_user_api_keys_max_life_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    before { SiteSetting.revoke_user_api_keys_maxlife_days = 2 }
+
+    it "should revoke the key" do
+      freeze_time
+
+      described_class.new.execute({})
+
+      expect(older_key.reload.revoked_at).to eq_time(Time.zone.now)
+      expect(newer_key.reload.revoked_at).to eq(nil)
+    end
```
Comment (jjaffeux) [inline spec/jobs/clean_up_user_api_keys_max_life_spec.rb:17]: For this kind of spec I think it's interesting to use the `change to` form:

```ruby
expect {
  described_class.new.execute({})
}.to change { older_key.revoked_at }.from(nil).to(eq_time(Time.zone.now))
 .and not_change { newer_key.revoked_at }
```

Didn't try it so might be wrong on syntax, but you get the idea

Comment (renato) [reply spec/jobs/clean_up_user_api_keys_max_life_spec.rb]: Thanks @jjaffeux! I've updated the tests with this approach and added the `active` filter to the existing user API clean up job to avoid updating `revoked_at` unnecessarily

**Review by davidtaylorhq — APPROVED** [review]
